src/aleph_client/commands/instance/network.py

```diff
@@ -76,23 +107,76 @@ async def fetch_crn_info(node_url: str) -> Optional[dict]:
+        firmware = safe_getattr(message, "content.environment.trusted_execution.firmware")
+        confidential = firmware and len(firmware) == 64
+        info = dict(
+            crn_hash=str(crn_hash) if crn_hash else "",
+            payment="hold\t   " if hold else str(safe_getattr(message, "content.payment.type.value")),
+            chain="Any" if hold else str(safe_getattr(message, "content.payment.chain.value")),
+            confidential=confidential,
+            allocation_type="",
+            ipv6_logs="",
+            crn_url="",
         )
-        raise InvalidURL("Invalid URL host")
-    return url
+        try:
+            # Fetch from the scheduler API directly if no payment or no receiver (hold-tier non-confidential)
+            if hold and not confidential:
+                try:
+                    info["allocation_type"] = help_strings.ALLOCATION_AUTO
+                    allocation = await fetch_json(
+                        session,
+                        f"https://scheduler.api.aleph.cloud/api/v0/allocation/{message.item_hash}",
+                    )
+                    nodes = await fetch_json(
+                        session,
+                        "https://scheduler.api.aleph.cloud/api/v0/nodes",
+                    )
+                    info["ipv6_logs"] = allocation["vm_ipv6"]
+                    for node in nodes["nodes"]:
+                        if node["ipv6"].split("::")[0] == ":".join(str(info["ipv6_logs"]).split(":")[:4]):
+                            info["crn_url"] = node["url"].rstrip("/")
+                    return message.item_hash, info
+                except Exception:
+                    info["ipv6_logs"] = help_strings.VM_SCHEDULED
+                    info["crn_url"] = help_strings.CRN_PENDING
+            else:
+                # Fetch from the CRN API if PAYG-tier or confidential
+                info["allocation_type"] = help_strings.ALLOCATION_MANUAL
+                for node in node_list.nodes:
+                    if node["hash"] == safe_getattr(message, "content.requirements.node.node_hash"):
+                        info["crn_url"] = node["address"].rstrip("/")
+                        path = f"{node['address'].rstrip('/')}/about/executions/list"
+                        executions = await fetch_json(session, path)
+                        if message.item_hash in executions:
+                            interface = IPv6Interface(executions[message.item_hash]["networking"]["ipv6"])
+                            info["ipv6_logs"] = str(interface.ip + 1)
+                            return message.item_hash, info
+                info["ipv6_logs"] = help_strings.VM_NOT_READY if confidential else help_strings.VM_NOT_AVAILABLE_YET
+        except (ClientResponseError, ClientConnectorError) as e:
+            info["ipv6_logs"] = f"Not available. Server error: {e}"
+        return message.item_hash, info
+
+
+async def find_crn_of_vm(vm_id: str) -> Optional[str]:
```

Comment:
Would it not be easier to use and test this function if it took the `client` in argument instead of creating one ?

Reply:
Where this function is used (inside 9 different commands), we don't instanciate any client, so not really needed and it will just lead to duplicated lines of code.